### PR TITLE
SONARIAC-545 Modify WorkdirInstruction implementation to use Argument

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java [hunk 1]
```diff
@@ -223,7 +223,10 @@ public WorkdirInstruction WORKDIR() {
     return b.<WorkdirInstruction>nonterminal(DockerLexicalGrammar.WORKDIR).is(
       f.workdir(
         b.token(DockerKeyword.WORKDIR),
-        ARGUMENTS()
+        b.oneOrMore(f.tuple(
+          b.token(DockerLexicalGrammar.WHITESPACE),
+          ARGUMENT()
+        ))
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java:229]: I can imagen having an `ARGUMENTS()` rule in the end which creates this list of tuples. For now, it's fine like this.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/WorkdirInstructionImpl.java [hunk 2]
```diff
@@ -22,23 +22,24 @@
...
 import org.sonar.iac.docker.tree.api.SyntaxToken;
 import org.sonar.iac.docker.tree.api.WorkdirInstruction;
 
 public class WorkdirInstructionImpl extends InstructionImpl implements WorkdirInstruction {
 
-  private final List<SyntaxToken> workdirList;
+  private final List<Argument> arguments;
 
-  public WorkdirInstructionImpl(SyntaxToken keyword, List<SyntaxToken> workdirList) {
+  public WorkdirInstructionImpl(SyntaxToken keyword, List<Argument> arguments) {
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/WorkdirInstructionImpl.java:33]: You could keep the variable name, to show what the arguments are actually representing in this context. But on the other hand, I could imagine an interface for such Instructions to provide the arguments() method. Could be useful for further processing. However, fine for now IMO.

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM! I've added some thoughts but no adjustments are needed.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java (hunk 1 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java:229]: 👍 For the first/two element I did it like this, but I will definitely create an `ARGUMENTS()` to handle the next ones (also will need to rename to old crap `ARGUMENTS()` :D )

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/WorkdirInstructionImpl.java (hunk 2 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/WorkdirInstructionImpl.java:33]: 👍 good point, I was also thinking that later we will probably have common interface/abstract class parent as we will have many instruction with this similar grammar `keyword argument+`

**Pull request merged**